src/BackupChain.cs

```diff
@@ -19,26 +20,33 @@ public class BackupChain : IBackupChain
   {
     private readonly IList<BackupMetadata> _orderedBackups;
 
+    /// <summary>
+    /// Chain needs to be 1 full backup, 1 diff and then the logs in the order they were taken
+    /// Sometimes backups can be striped (i.e. split into multiple files) - so we need to handle these cases too
+    /// https://www.sqlservercentral.com/articles/getting-a-list-of-the-striped-backup-files
+    /// </summary>
+    /// <param name="recentBackups"></param>
     private BackupChain(IList<BackupMetadata> recentBackups)
     {
       var backups = recentBackups.Distinct(new BackupMetadataEqualityComparer())
-        .Where(b => IsValidFilePath(b)) // A third party application caused invalid path strings to be inserted into backupmediafamily
+        .Where(IsValidFilePath) // A third party application caused invalid path strings to be inserted into backupmediafamily
         .ToList();
 
-      var mostRecentFullBackup = MostRecentFullBackup(recentBackups);
-      _orderedBackups = new List<BackupMetadata> { mostRecentFullBackup };
-
-      var differentialBackup = MostRecentDifferentialBackup(backups, mostRecentFullBackup);
-      if(differentialBackup != null)
-        _orderedBackups.Add(differentialBackup);
-
-      var mostRecentBackup = _orderedBackups.Last();
-      while(mostRecentBackup != null) {
-        mostRecentBackup = NextLogBackup(backups, mostRecentBackup);
-
-        if(mostRecentBackup != null)
-          _orderedBackups.Add(mostRecentBackup);
+      var mostRecentFullBackups = MostRecentFullBackups(recentBackups).ToList();
+      
+      var differentialBackups = MostRecentDifferentialBackups(backups, mostRecentFullBackups.First());
+      // differentialBackups can be null
+      
+      var orderedBackups = mostRecentFullBackups.Concat(differentialBackups).ToList();

```

Comment:
Why'd you change this logic?
You can use `AddRange` instead of `Add`

Reply:
ahh sorry, residual from when I was using `IEnumerable` everywhere
Will change this